### PR TITLE
Warning free ghc 7.10

### PR DESCRIPTION
**Pull request by phadej**
Example travis build: https://travis-ci.org/phadej/monad-time/builds/89182592

### CODE REVIEW

File: monad-time.cabal [hunk 1]
```diff
@@ -20,6 +26,13 @@ library
 
   hs-source-dirs:    src
 
-  ghc-options:       -O2 -Wall -funbox-strict-fields
+  ghc-options:       -Wall
```
Comment (phadej) [inline monad-time.cabal:29]: There are no data definitions, and `-O2` is unneeded here. makes `cabal check` way happier.

**Comment by arybczak** [comment]
Thanks for the patch. How strongly do you feel about ifdefs in Control.Monad.Time.Instances though? I feel like putting OPTIONS_GHC -fno-warn-deprecated-flags there would achieve the same effect and be much less noisy.

**Comment by phadej** [comment]
@arybczak If you are ok with having `-fno-warn-deprecated-flags` (i.e. what we might miss in the future) then I can change the PR.

**Comment by phadej** [comment]
@arybczak I turned the pragma for older GHC's and added GHC8.0.1 into build matrix.

**Comment by arybczak** [comment]
If you add -fno-warn-deprecated-flags (and leave OverlappingInstances), then you do not have to add OVERLAPPABLE nor any additional ifdefs checking for GHC version :) GHC 8.0.1 still supports OverlappingInstances I presume?

**Comment by arybczak** [comment]
Merged (with a few modifications).

**Comment by phadej** [comment]
Well, in your case using of `OverlappingInstances` is quite safe, as there aren't other instances which can be overlapped. The culprit there is, that `LANGUAGE OverlappingInstances` set's all instances as both _overlappable_ and _overlapping_ which might be problematic. But as said, I don't see that your approach could be problematic, as I guess you don't have plans to add more instances.

**Pull request closed without merging**